### ISSUE TITLE
Using CSS preprocessors

### ISSUE THREAD

**Issue by ackzell** [jsreport/jsreport#300]
I tried searching for a reference that a preprocessor such as Sass or LESS would be supported, but not sure that it is the case.

The idea is to instead of writing an asset which is written in pure CSS, I would feed it a `.scss` file and this would be transformed into CSS. Where would this functionality fit in on the platform?

Could you clarify?

**Comment by pofider** [comment]
Thank you for the question.

It sounds to me that such thing shouldn't be part of the jsreport scope. You should do the compilation on your own as a pre-deployment step.

Maybe you can try to explain why you want to have this directly inside the jseport? You could solve it with a custom extension, but so far I don't see it as a job for jsreport.

**Comment by ackzell** [comment]
Thanks for the quick reply, as usual @pofider :)

Yes, I agree it doesn't feel like something jsreport should do but maybe I should have referred to _jsreport Studio_ and ask the question on that repo.

I am currently exploring the phantomjs recipe to create a PDF file, and find myself missing variables, mixins, and nesting on the CSS because that would make my life easier at the moment.

The idea is to leverage a preprocessor during _report layout implementation_. I am thinking of a workflow where the editors in studio allow you to use different preprocessors like when you choose in between engines.

Stating the obvious a little bit: making use of these would allow for a faster iteration on implementing styles and thus, being able to see what the output would look like quicker.

**Comment by bjrmatos** [comment]
> The idea is to leverage a preprocessor during report layout implementation. I am thinking of a workflow where the editors in studio allow you to use different preprocessors like when you choose in between engines.
> 
> Stating the obvious a little bit: making use of these would allow for a faster iteration on implementing styles and thus, being able to see what the output would look like quicker.

interesting, i also think it does not belong in jsreport, but technically if jsreport-assets extension can provide a way to define custom encodings this can be easily solved with third-party code (at the cost of adding some overhead to the rendering process because it will compile files on pdf render but can be solved with a cache or just pre-compiling everything).

thinking about your use case i think that you can setup less/sass/stylus/postcss, etc normally in your project, create a folder (something like `stylesSrc`), use your favorite pre/post-processor to watch that directory and compile on every change to source files, then you just need to reference the build output (css files)) of your favorite pre/post-processor as a local asset (with a local path), it does not sound too bad but i can see that a integrated workflow in jsreport studio could be useful too.

**Comment by ackzell** [comment]
Thanks!
Yes, that makes a lot of sense. 

I'll be playing with the local path assets again. I tried but didn't succeed in grabbing the files in that manner.

Anyway, I will try and follow your advise @bjrmatos ,

cheers!

**Comment by ackzell** [comment]
Figured it! I just needed to be more careful following instructions. 

Indeed it isn't so bad to have a _decoupled compile CSS phase_ in a workflow since the asset referenced from studio is synced up seamlessly (or that is at least what some fiddling has proven so far 😁 ). Also when working on the Front End one is kind of used to the whole "setup a task runner for your needs" type of workflow. 

>  i can see that a integrated workflow in jsreport studio could be useful too.
Either by an extension on jsreport (studio), or having some kind of integration for particular editors, like extensions or plugins. I'm just ranting here, but what I found is that I feel I am more productive by opening up the editor I'm more used to using and creating/editing the files I need there.

Thanks a lot BJR, and Jan.
